code/modules/surgery/robotics.dm

```diff
@@ -428,3 +431,146 @@
+
+/datum/surgery_step/robotics/cavity/implant_removal
+	allowed_tools = list(
+	/obj/item/weapon/hemostat = 100,	\
+	/obj/item/weapon/wirecutters = 75,	\
+	/obj/item/weapon/material/kitchen/utensil/fork = 20
+	)
+
+	min_duration = 80
+	max_duration = 100
+
+	can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+		var/obj/item/organ/brain/sponge = target.internal_organs_by_name["brain"]
+		return ..() && (!sponge || !sponge.damage)
+
+	begin_step(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+		var/obj/item/organ/external/affected = target.get_organ(target_zone)
+		user.visible_message("[user] starts poking around inside [target]'s [affected.name] with \the [tool].", \
+		"You start poking around inside [target]'s [affected.name] with \the [tool]" )
+		..()
+
+	end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+		var/obj/item/organ/external/chest/affected = target.get_organ(target_zone)
+
+		var/find_prob = 0
+
+		if (affected.implants.len)
+
+			var/obj/item/obj = pick(affected.implants)
+
+			if(istype(obj,/obj/item/weapon/implant))
+				var/obj/item/weapon/implant/imp = obj
+				if (imp.islegal())
+					find_prob +=60
+				else
+					find_prob +=40
+			else
+				find_prob +=50
+
+			if (prob(find_prob))
+				user.visible_message("<span class='notice'>[user] takes something out of incision on [target]'s [affected.name] with \the [tool].</span>", \
+				"<span class='notice'>You take [obj] out of incision on [target]'s [affected.name]s with \the [tool].</span>" )
+				affected.implants -= obj
+
+				BITSET(target.hud_updateflag, IMPLOYAL_HUD)
+
+				//No need to handle borers, IPCs only.
+				obj.loc = get_turf(target)
+				obj.add_blood(target)
+				obj.update_icon()
+				if(istype(obj,/obj/item/weapon/implant))
+					var/obj/item/weapon/implant/imp = obj
+					imp.imp_in = null
+					imp.implanted = 0
+			else
+				user.visible_message("<span class='notice'>[user] removes \the [tool] from [target]'s [affected.name].</span>", \
+				"<span class='notice'>There's something inside [target]'s [affected.name], but you just missed it this time.</span>" )
+		else
+			user.visible_message("<span class='notice'>[user] could not find anything inside [target]'s [affected.name], and pulls \the [tool] out.</span>", \
+			"<span class='notice'>You could not find anything inside [target]'s [affected.name].</span>" )
+
+	fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+		..()
+		var/obj/item/organ/external/chest/affected = target.get_organ(target_zone)
+		if (affected.implants.len)
+			var/fail_prob = 10
+			fail_prob += 100 - tool_quality(tool)
+			if (prob(fail_prob))
+				var/obj/item/weapon/implant/imp = affected.implants[1]
+				user.visible_message("<span class='warning'>Something beeps inside [target]'s [affected.name]!</span>")
+				playsound(imp.loc, 'sound/items/countdown.ogg', 75, 1, -3)
+				spawn(25)
```

Comment:
Timer!

Reply:
This was copied from the implant file, so a timer would need to be used there, too. What's the proper syntax for it?

Reply:
Resolved.